src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts

```diff
@@ -362,4 +358,12 @@ export class RgwBucketFormComponent extends CdForm implements OnInit {
   getMfaDeleteStatus() {
     return this.isMfaDeleteEnabled ? RgwBucketMfaDelete.ENABLED : RgwBucketMfaDelete.DISABLED;
   }
+
+  private getLockDays(bucketData: object): number {
+    if (bucketData['lock_retention_period_years'] > 0) {
+      return Math.floor(bucketData['lock_retention_period_years'] * 365.242);
```

Comment:
What if some user create bucket from CLI passing both days and years (which would be `days+ years*365` total no. of days, right?) then shouldn't we consider both in calculation? So the return statement should be like this
`Math.floor(bucketData['lock_retention_period_years'] * 365.242) + bucketData['lock_retention_period_days'];`

Reply:
@avanthakkar The backend API does not allow to set both: either you get days or years.